aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -89,6 +91,36 @@ impl TransactionMetadata {
                 TransactionPayload::Multisig(m) => Some(m.clone()),
                 _ => None,
             },
+            raw_transaction_hash: match txn.authenticator_ref() {
+                MultiAgent {
+                    sender: _,
+                    secondary_signer_addresses,
+                    secondary_signers: _,
+                } => {
+                    let raw_txn = RawTransactionWithData::new_multi_agent(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                    );
+                    raw_txn.hash().to_vec()
+                },
+                FeePayer {
+                    sender: _,
+                    secondary_signer_addresses,
+                    secondary_signers: _,
+                    fee_payer_address: _,
+                    fee_payer_signer: _,
+                } => {
+                    // In the case of a fee payer transaction, the hash value is generated using
+                    // AccountAddress::ZERO as the fee payer address.
+                    let raw_txn = RawTransactionWithData::new_fee_payer(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                        AccountAddress::ZERO,
+                    );
+                    raw_txn.hash().to_vec()
+                },
+                _ => txn.raw_transaction_ref().hash().to_vec(),
```

Comment:
`SignedTransaction` actually already has the field `committed_hash` that is computed during the previous stages of the execution pipeline. If we are okay with using `hash(signed transaction)` here instead of `hash(raw transaction)`, then we might save CPU cycles?

Reply:
We need the hash of the "raw transaction" that is used in the authentication process for signers.

Reply:
If we are okay with using `hash(signed transaction)`, then I guess we may not need to construct `RawTransactionWithData` as above (as TransactionAuthenticator will already have the info)?